File: codechef/december2017/CPLAY.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long
int main() {
    string s;
    while(cin>>s)
    {
        int i,j,k,l,m,g1=0,g2=0;
        string a,b;
        for(i=0;i<20;i+=2)
        {
            a=a+s[i];
        }
        for(i=1;i<20;i+=2)
        {
            b=b+s[i];
        }
        int flag=0;
        for(i=0,j=0;i<5;)
        { if((g1>g2+5-j)||(g2>g1+5-i))
        {flag=1;

            break;
        }

            if(a[i++]=='1')
            { g1++;

            }
            if((g1>g2+5-j)||(g2>g1+5-i))
            {flag=1;

                break;
            }

            if(b[j++]=='1')
            {g2++;
            }
            if((g1>g2+5-j)||(g2>g1+5-i))
            {flag=1;

                break;
            }
        }

        if(flag==1 && g1!=g2)
        {
            if(g1>g2)
                cout<<"TEAM-A "<<i+j<<endl;
            else
                cout<<"TEAM-B "<<i+j<<endl;
        }
        else
        {
            for(i=5;i<10;i++)
            {if(a[i]=='1'&&b[i]=='0')
            {
                cout<<"TEAM-A "<<2*i+2<<endl;
                break;
            }
                if(a[i]=='0'&&b[i]=='1')
                {
                    cout<<"TEAM-B "<<2*i+2<<endl;
                    break;
                }
            }
            if(i==10)
            {cout<<"TIE"<<endl;}

        }

    }

}

File: codechef/december2017/CPLAY.java
import java.util.*;
class Main{

  public static void main(String[] args) {
    String s;
    Scanner in = new Scanner(System.in);

    while(in.hasNext()){
      s = in.next();
      int m,p1=0,p2=0;
      String a =new String();
      String b =new String();
      for(int i=0;i<20;i+=2){
        a = a+s.charAt(i);
      }

      for(int i=1; i<20 ; i+=2){
        b = b+s.charAt(i);
      }

      int flag = 0;
      int i=0,j=0;
      for(i=0,j=0;i<5;){
        if(( p1>p2+5-j) || ( p2>p1 + 5 -i)){
          flag = 1;
          break;
        }

        if(a.charAt(i++)=='1'){
          p1++;
        }
        if((p1 > p2+5-j)|| (p2>p1+5-i)){
          flag = 1;
          break;
        }

        if(b.charAt(j++)=='1'){
          p2++;
        }
        if((p1>p2+5-j)||(p2>p1+5-i)){
          flag=1;
          break;
        }
      }

      if(flag==1 && p1!=p2){
          if(p1>p2)
              System.out.print("TEAM-A "+ (i+j) + "\n");
          else
              System.out.print("TEAM-B "+(i+j)+"\n");
      }else{

          for(i=5;i<10;i++){
            if(a.charAt(i)=='1' && b.charAt(i)=='0'){
              System.out.print("TEAM-A "+(2*i+2)+"\n");
              break;
            }

            if(a.charAt(i)=='0' && b.charAt(i)=='1'){
              System.out.print("TEAM-B "+(2*i+2)+"\n");
              break;
            }
          }
          if(i==10){
            System.out.print("TIE\n");
          }
      }
    }
  }
}
